docs/concepts/ksql-architecture.rst

```diff
@@ -0,0 +1,371 @@
+    changing, or deleting streams and tables. Data Definition Language
+    statements modify metadata only and don't operate on data. You can use
+    these statements with declarative DML statements.
+
+    The DDL statements include:
+
+    * CREATE STREAM 
+    * CREATE TABLE
+    * DROP STREAM
+    * DROP TABLE
+
+Data Manipulation Language (DML) Statements
+    Declarative verbs that read and modify data in KSQL streams and tables.
+    Data Manipulation Language statements modify data only and don't change
+    metadata. The KSQL engine compiles DML statements into Kafka Streams
+    applications, which run on a Kafka cluster like any other Kafka Streams
+    application.
+
+    The DML statements include:
+
+    * CREATE STREAM AS SELECT (CSAS) 
+    * CREATE TABLE AS SELECT (CTAS)
+    * SELECT
+    * INSERT INTO
+
+For more information, see :ref:`ksql_syntax_reference`.
+
+.. _ksql-deployment-modes:
+
+KSQL Deployment Modes
+*********************
+
+KSQL supports three ways to deploy your streaming applications:
+
+* Interactive – data exploration and pipeline development
+* Headless – long-running production environments
+* Embedded – KSQL statements and queries run directly in your Java
+  applications
+
+In all deployment modes, KSQL enables distributing the processing load for your
+KSQL applications across all KSQL server instances, and scaling up and down
+without restarting your applications.
+
+Interactive Deployment
+====================== 
+
+Use the interactive mode to develop your KSQL applications. When you deploy a
+KSQL server in interactive mode, the REST interface is available for the KSQL
+CLI and |c3| to connect to. 
+
+.. image:: ../img/ksql-client-server-interactive-mode.png
+   :alt: Diagram showing interactive KSQL deployment
+   :align: center
+
+In interactive mode, you can:
+
+* Write statements and queries on the fly
+* Start any number of server nodes: ``bin/ksql-server-start``
+* Start one or more CLIs or REST Clients and point them to a server: ``bin/ksql https://myksqlserver:8090``
+
+Headless Deployment
+===================
+
+Use headless mode to deploy your KSQL application to a production environment.
+When you deploy a KSQL server in headless mode, the REST interface isn't
+available, so you assign workloads to KSQL servers by using a SQL file. The SQL
+file contains the KSQL statements and queries that define your application.
+Headless mode is ideal for streaming ETL application deployments.
+
+.. image:: ../img/ksql-standalone-headless.png
+   :alt: Diagram showing headless KSQL deployment
+   :align: center
+
+In headless mode you can:
+
+* Start any number of server nodes
+* Pass a SQL file with KSQL statements to execute: ``bin/ksql-node query-file=foo/bar.sql``
+* Version-control your queries and transformations as code
+* Ensure resource isolation
+* Leave resource management to dedicated systems, like Kubernetes
+
+Embedded Deployment

```

Comment:
Is embedded mode now an officially supported execution mode for KSQL?

Reply:
Good catch. It is not (see https://github.com/confluentinc/ksql/issues/734).  Until we support it officially, I wouldn't mention it in our docs. cc @MichaelDrogalis

Reply:
+1, hold until it's shipped.